app/assets/javascripts/discourse/app/controllers/preferences/interface.js

```diff
@@ -185,6 +185,48 @@ export default class InterfaceController extends Controller {
     return userOptionTextSize !== selectedTextSize;
   }
 
+  _isInLightMode() {
+    return (
+      this.interfaceColor.colorModeIsLight ||
+      (this.interfaceColor.colorModeIsAuto &&
+        !window.matchMedia("(prefers-color-scheme: dark)").matches)
+    );
+  }
+
+  _isInDarkMode() {
+    return (
+      this.interfaceColor.colorModeIsDark ||
+      (this.interfaceColor.colorModeIsAuto &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches)
+    );
+  }
+
+  _shouldEnablePreview(isDarkMode) {
+    return (
+      this.isViewingOwnProfile &&
+      (isDarkMode ? this._isInDarkMode() : this._isInLightMode())
+    );
+  }
+
+  _resolveThemeDefaultColorScheme(colorSchemeId, isDark = false) {
+    if (!isDark && colorSchemeId >= 0) {
+      return colorSchemeId;
+    }
+    if (isDark && colorSchemeId !== -1) {
+      return colorSchemeId;
+    }
```

Comment:
could have a brief comment for each of those `if`s that explains what 0 and -1 are in this context?